src/v/compat/cluster_json.h

```diff
@@ -666,6 +673,12 @@ inline void read_value(json::Value const& rd, cluster::topic_properties& obj) {
       rd,
       "initial_retention_local_target_ms",
       obj.initial_retention_local_target_ms);
+    std::optional<ss::sstring> vcluster_str;
+    read_member(rd, "virtual_cluster_id", vcluster_str);
```

Comment:
question: Should this be `mpx_virtual_cluster_iod`?

Reply:
If you apply the suggestion above you can do `read_member(rd, "mpx_virtual_cluster_id", obj.mpx_virtual_cluster_id)`